Email Pattern accepts double dot between domain and tld

### Description of the bug

I'm using an EmailField without overriding the  validation pattern.
An email like "example@gmail..com" is not considered invalid (using `emailField.isInvalid()`)

I got an exception from javax.mail.internet.InternetAddress when I send an email to this address:

```
Caused by: javax.mail.internet.AddressException: Domain contains dot-dot
at javax.mail.internet.InternetAddress.checkAddress(InternetAddress.java:1413) ~[jakarta.mail-1.6.7.jar!/:1.6.7]
at javax.mail.internet.InternetAddress.parse(InternetAddress.java:1191) ~[jakarta.mail-1.6.7.jar!/:1.6.7]
at javax.mail.internet.InternetAddress.parse(InternetAddress.java:728) ~[jakarta.mail-1.6.7.jar!/:1.6.7]
at javax.mail.internet.InternetAddress.parse(InternetAddress.java:705) ~[jakarta.mail-1.6.7.jar!/:1.6.7]
```


### Expected behavior

The method `isInvalid()` should return true when email address contains multiple dots (like example@gmail..com)

### Minimal reproducible example

```java
var emailField = new EmailField("Email");
emailField.setValue("example@gmail..com");
Notification.show(emailField.getValue() + " : " + emailField.isInvalid());
```

### Versions

- Vaadin / Flow version: 23.1.2 
- Java version: 17
- OS version: Docker image from eclipse-temurin:17-jdk-focal
- Browser version (if applicable): 
- Application Server (if applicable): Embedded Tomcat with Spring Boot 2.6.7
- IDE (if applicable):

--- a/flow-data/src/main/java/com/vaadin/flow/data/validator/EmailValidator.java
+++ b/flow-data/src/main/java/com/vaadin/flow/data/validator/EmailValidator.java
@@ -26,9 +26,9 @@ package com.vaadin.flow.data.validator;
 @SuppressWarnings("serial")
 public class EmailValidator extends RegexpValidator {
 
-    private static final String PATTERN = "^" + "([a-zA-Z0-9_\\.\\-+])+" // local
-            + "@" + "[a-zA-Z0-9-.]+" // domain
-            + "\\." + "[a-zA-Z0-9-]{2,}" // tld
+    public static final String PATTERN = "^" + "([a-zA-Z0-9_\\.\\-+])+" // local
+            + "@" + "([a-zA-Z0-9-]+\\.)+" // domain with the dot separator
+            + "[a-zA-Z0-9-]{2,}" // before tld
             + "$";
 
     private final boolean allowEmptyValue;

--- a/flow-data/src/test/java/com/vaadin/flow/data/validator/EmailValidatorTest.java
+++ b/flow-data/src/test/java/com/vaadin/flow/data/validator/EmailValidatorTest.java
@@ -100,6 +100,13 @@ public class EmailValidatorTest extends ValidatorTestBase {
                 false));
     }
 
+    @Test
+    public void testDomainWithDotDotFails() {
+        EmailValidator v = validator("domains containing dot dot should fail");
+        assertFails("hello@sample..com", v);
+        assertFails("hello@samp..le..com", v);
+    }
+
     private EmailValidator validator(String errorMessage) {
         return new EmailValidator(errorMessage);
     }

